### PR TITLE
[ENH]  Added `FormControlLabel` for clickable checkbox text

### LINKED ISSUE #58: [ENH] Make text associated with checkbox clickable
### Is there an existing issue for this?

- [X] I have searched the existing issues

### New feature

Small quality-of-life feature, checkboxes are small, make it so the text associated with them also check/uncheck the box.

"Select all" for example

### Unclear documentation

_No response_

### PR DESCRIPTION
**Pull request by samadpls**
- Closes #58 

Changes proposed in this pull request:

- Added `FormControlLabel` for clickable checkbox text

## Checklist

_This section is for the PR reviewer_

- [X] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see our [Contributing Guidelines](https://neurobagel.org/contributing/CONTRIBUTING#pull-request-guidelines) for more info)_
- [ ] PR has a label for the release changelog or `skip-release` (to be applied by maintainers only)
- [X] PR links to GitHub issue with mention `Closes #XXXX`
- [ ] Tests pass
- [ ] Checks pass

For new features:

- [ ] Tests have been added

For bug fixes:

- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ The first release of the tool refactored and reimplemented in React and TypeScri
 - [MNT] Added post-merge hook and updated scripts [#38](https://github.com/neurobagel/query-tool/pull/38) ([@Sauradip07](https://github.com/Sauradip07))
 - [CI] Set up GitHub pages deployment CI [#50](https://github.com/neurobagel/query-tool/pull/50) ([@rmanaem](https://github.com/rmanaem))
 - [CI] Set up ESLint CI [#47](https://github.com/neurobagel/query-tool/pull/47) ([@rmanaem](https://github.com/rmanaem))
-- [CI]  Implement Prettier CI Check for Code Consistency [#44](https://github.com/neurobagel/query-tool/pull/44) ([@samadpls](https://github.com/samadpls))
+- [CI] Implement Prettier CI Check for Code Consistency [#44](https://github.com/neurobagel/query-tool/pull/44) ([@samadpls](https://github.com/samadpls))
```
Comment (surchs) [inline CHANGELOG.md]: This is an autogenerated file. Could you please roll this back - I added it to the prettierignore file in #83

File: src/components/ResultCard.tsx [hunk 2]
```diff
@@ -31,10 +32,15 @@ const ResultCard = memo(
...
-              onChange={() => onCheckboxChange(datasetUUID)}
+            <FormControlLabel
+              control={
+                <Checkbox
+                  data-cy={`card-${datasetUUID}-checkbox`}
+                  checked={checked}
+                  onChange={() => onCheckboxChange(datasetUUID)}
+                />
+              }
+              label={datasetName}
```
Comment (surchs) [inline src/components/ResultCard.tsx:43]: This is a nice fix, but we don't want to add a label next to the checkbox, especially not with the `datasetName` that you can already see on the dataset card. So I think we need to talk a bit more about what to do here. From a UX perspective, the nicest thing might be to make the entire card element clickable to set and unset the checkbox.

**Review by surchs — CHANGES_REQUESTED** [review]
Thanks @samadpls for the PR - but I think we need to talk a bit more about what the goal is here. We don't want to add a new label just to make the checkbox work. Maybe @rmanaem has some ideas here as well. But the main goal from a user perspective is "I can click outside the checkbox and the checkbox will still become checked". That can be achieved in a number of ways. Maybe you have some ideas?

**Comment by surchs** [comment]
@samadpls would you like to either discuss or propose a solution here without adding a new label to the checkbox? Essentially without this:
image

**Comment by samadpls** [comment]
> @samadpls would you like to either discuss or propose a solution here without adding a new label to the checkbox? Essentially without this:
> image

I agree with your suggestion to avoid adding a new label to the checkbox. Perhaps we can add an `onClick` event to the Card component or explore other options you recommend.

**Comment by rmanaem** [comment]
> I agree with your suggestion to avoid adding a new label to the checkbox. Perhaps we can add an onClick event to the Card component or explore other options you recommend.

Let's try the `onClick` event and explore other solutions if that doesn't work out as expected.

**Comment by samadpls** [comment]
Apologies for any inconvenience caused. Due to time constraints, I won't be able to follow up on this specific PR.😢
However, It was a great experience working with all of you during the GSoC selection process!😊

**Comment by rmanaem** [comment]
No problem @samadpls, It's understandable. Thanks for contributing to the project! 
Please stay in touch and feel welcome to contribute again to our repos!

**Pull request closed without merging**